src/lib/Tests/Pagination/Pagerfanta/URLWildcardAdapterTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Tests\Pagination\Pagerfanta;
+
+use eZ\Publish\API\Repository\URLWildcardService;
+use eZ\Publish\API\Repository\Values\Content\UrlWildcard;
+use Ibexa\AdminUi\Pagination\Pagerfanta\URLWildcardAdapter;
+use PHPUnit\Framework\TestCase;
+
+class URLWildcardAdapterTest extends TestCase
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\UrlWildcard|\PHPUnit\Framework\MockObject\MockObject */
+    private $urlWildcardService;
+
+    protected function setUp(): void
+    {
+        $this->urlWildcardService = $this->createMock(URLWildcardService::class);
+    }
+
+    public function testGetNbResults()
+    {
+        $countAll = 5;
+
+        $this->urlWildcardService
+            ->expects($this->once())
+            ->method('countAll')
+            ->with()
+            ->willReturn($countAll);
+
+        $adapter = new URLWildcardAdapter($this->urlWildcardService);
+
+        $this->assertEquals(
+            $countAll,
+            $adapter->getNbResults()
+        );
+    }
+
+    public function testGetSlice()
+    {
+        $offset = 10;
+        $limit = 25;
+
+        $this->urlWildcardService
+            ->expects($this->once())
+            ->method('loadAll')
+            ->with($offset, $limit)
+            ->willReturn([$this->urlWildcards()]);
+
+        $adapter = new URLWildcardAdapter($this->urlWildcardService);
+
+        $this->assertEquals(
+            [$this->urlWildcards()],
+            $adapter->getSlice($offset, $limit)
+        );
```

Comment:
I understand what you want to do here, but in this case the comparison will simply be wrong (albeit it passes tests because the Generators that are actually the only element of the array are identical).

In actuality, the array being compared is:
```php
[
    <Generator>,
];
```
instead of the expected:
```php
[
    <UrlWildcard>,
    <UrlWildcard>,
];
```

What you can do is unpack the Generator:
```suggestion
        $expectedResult = iterator_to_array($this->urlWildcards());
        $this->urlWildcardService
            ->expects($this->once())
            ->method('loadAll')
            ->with($offset, $limit)
            ->willReturn($expectedResult);

        $adapter = new URLWildcardAdapter($this->urlWildcardService);

        $this->assertEquals(
            $expectedResult,
            $adapter->getSlice($offset, $limit)
        );
```
or drop the Generator (`iterable`) entirely and return a simple array in `urlWildcards` method.